querybook/webapp/ui/Comment/Reactions.tsx

```diff
@@ -0,0 +1,106 @@
+import clsx from 'clsx';
+import * as React from 'react';
+import { useSelector } from 'react-redux';
+
+import { IStoreState } from 'redux/store/types';
+import { StyledText } from 'ui/StyledText/StyledText';
+
+import { AddReactionButton } from './AddReactionButton';
+import { IReaction } from './Comments';
+
+interface IProps {
+    reactions: IReaction[];
+}
+
+export const Reactions: React.FunctionComponent<IProps> = ({
+    reactions: reactionsProp,
+}) => {
+    const [uidsByReaction, setUidsByReaction] = React.useState<
+        Record<string, number[]>
+    >({});
+    const userInfo = useSelector((state: IStoreState) => state.user.myUserInfo);
+
+    const formatReactions = React.useCallback((reactions: IReaction[]) => {
+        const formattedReactions = {};
+        for (const reaction of reactions) {
+            if (reaction.reaction in formattedReactions) {
+                formattedReactions[reaction.reaction] = [
+                    ...formattedReactions[reaction.reaction],
+                    reaction.uid,
+                ];
+            } else {
+                formattedReactions[reaction.reaction] = [reaction.uid];
+            }
+        }
+        return formattedReactions;
+    }, []);
+
+    React.useEffect(() => {
+        if (reactionsProp) {
+            setUidsByReaction(formatReactions(reactionsProp));
```

Comment:
i image in future this will be
```
const uidsByReaction = useMemo(() => formatReactions(reactionsProps), [formatReactions, reactionsProps]);
```
and handleReactionClick would update the parent component instead?